drivers/i3c/i3c_dw.c

```diff
@@ -680,19 +680,15 @@ static int get_i3c_addr_pos(const struct device *dev, uint8_t addr, bool sa)
  * utilizes the controller private data for where the id reg is stored.
  *
  * @param dev Pointer to the I3C device structure.
- * @param addr I2C address of the device whose position is to be retrieved.
+ * @param desc Pointer to target device descriptor.
  *
- * @return The position (ID) of the device on success, or a negative error code
- *         if the device with the given address is not found.
+ * @return The position (ID) of the device
  */
-static int get_i2c_addr_pos(const struct device *dev, uint16_t addr)
+static int get_i2c_addr_pos(const struct device *dev, struct i3c_i2c_device_desc *desc)
 {
 	struct dw_i3c_i2c_dev_data *dw_i3c_device_data;
-	struct i3c_i2c_device_desc *desc = i3c_dev_list_i2c_addr_find(dev, addr);
```

Comment:
I'm not sure why this was changed, This could allow it to talk to a i2c device without calling the `i3c_attach_i2c_device` first, which shouldn't happen.

With that `attach_i2c_device` API removed, there is still some 'stuff' that happens in `i3c_attach_i2c_device` that is neccessary

Reply:
This function is only called in one place ```dw_i3c_i2c_transfer``` right after we have called ```dw_i3c_i2c_attach_device```, I probably should just remove it and obtain the dynamically selected position there instead.